src/renderer/Dropdown.tsx

```diff
@@ -0,0 +1,127 @@
+import {
+  createRef,
+  KeyboardEvent,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react';
+import ArrowIcon from './icons/ArrowIcon';
+import './styles/Dropdown.scss';
+import { useClickOutside, useFocusOutside } from './utils';
+
+interface IOptionEntry {
+  value: string;
+  label: string;
+  display: JSX.Element | string;
+}
+
+interface IDropdownProps {
+  name: string;
+  options: IOptionEntry[];
+  value: string;
+  handleChange: (newValue: string) => void;
+}
+
+const Dropdown = ({ name, options, value, handleChange }: IDropdownProps) => {
+  const inputRefs = useMemo(
+    () =>
+      Array(options.length)
+        .fill(0)
+        .map(() => createRef<HTMLLIElement>()),
+    [options]
+  );
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const labelRef = useRef<HTMLLabelElement>(null);
+
+  useEffect(() => {
+    if (isOpen) {
+      const index = options.findIndex((entry) => entry.value === value);
+      inputRefs[index].current?.focus();
+    }
+  }, [inputRefs, isOpen, options, value]);
+
+  useClickOutside<HTMLLabelElement>(labelRef, () => {
+    setIsOpen(false);
+  });
+
+  useFocusOutside<HTMLLabelElement>(labelRef, () => {
```

Comment:
Yup, you're right.

Note: we can't remove the `useClickOutside` hook in favour of `useFocusOutside` because a user could click on a div outside of the dropdown and that doesn't change which element is in focus. It would only work if a user clicked on a different input field.